### ISSUE TITLE
Identify and warn you when RUN instructions have a dangling backslash

### ISSUE THREAD

**Issue by nickjj** [hadolint/hadolint#973]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

A dangling backslash in a `RUN` instruction can cause future instructions to fail with misleading errors.

For example:

```Dockerfile
FROM debian:bookworm-slim AS base

RUN echo "Base..."

FROM base AS dependencies

RUN apt-get update -y \
      && apt-get install -y \
           curl \
           git \

FROM dependencies AS test

RUN echo "Test..."

FROM test AS build

CMD ["echo", "Done!"]
```

This will not build because of the `git \` line near the middle. It should really be `git`.

As is you'll get this error from Docker `=> ERROR [internal] load metadata for docker.io/library/test:latest`.

### Proposed solution

One possibility is if you're within a `RUN` command and the line ends with `\` and the next non-comment word in the file is a valid Docker instruction then produce a lint warning that you have a dangling backslash.

For example in the above case, we had `\` followed by new lines but then the next "real" word was `FROM` which is a valid Docker instruction.

I'll admit I didn't think through the solution in great detail but it feels parse'able?
